### PR TITLE
Resolve dependency upgrade lint fixes

### PR DESCRIPTION
**Pull request by taysea**
[Deploy Preview](https://deploy-preview-INSERT_PR_#_HERE--keen-mayer-a86c8b.netlify.app/)

#### What does this PR do?
Fixes eslintrc and files to adjust to upgraded dependencies. Some new defaults were disabled (to align with Grommet disabled rules) and some code was adjusted to fix errors.

#### What testing has been done on this PR?

In addition to the feature you are implementing, have you checked the following:

**General UX Checks**
- [ ] Small, medium, and large screen sizes
- [ ] Cross-browsers (FireFox, Chrome, and Safari)
- [ ] Light & dark modes
- [ ] All hyperlinks route properly

**Accessibility Checks**
- [ ] Keyboard interactions
- [ ] Screen reader experience
- [ ] Run WAVE accessibility plugin (Chrome)

**Code Quality Checks**
- [ ] Console is free of warnings and errors
- [ ] Passes E2E commit checks
- [ ] Visual snapshots are reasonable

### CODE REVIEW

File: aries-site/src/examples/components/datatable/DataTableMultiSelectExample.js [hunk 1]
```diff
@@ -317,33 +317,33 @@ TableControls.propTypes = {
...
-            </Text>
-            <Text size="small">items</Text>
-          </Box>
-        )}
-      </>
-    )}
-  </>
-);
+const SelectionSummary = ({ selected }) => {
+  if (selected) {
```
Comment (halocline) [inline aries-site/src/examples/components/datatable/DataTableMultiSelectExample.js:321]: I don't think we want this change. Should disable this rule.

File: aries-site/src/examples/components/namevaluelist/Examples/NameValueListDefaultExample.js [hunk 2]
```diff
@@ -16,7 +16,7 @@ export const NameValueListDefaultExample = () => (
         }
         return (
           <NameValuePair key={name} name={name}>
-            {name === 'Created on' ? <> {date}</> : <>{value}</>}
+            {name === 'Created on' ? <> {date}</> : value}
```
Comment (halocline) [inline aries-site/src/examples/components/namevaluelist/Examples/NameValueListDefaultExample.js:19]: confirming that this is a "no useless fragments" rule?

File: aries-site/src/examples/components/namevaluelist/Examples/NameValueListWizard.js [hunk 3]
```diff
@@ -100,24 +100,28 @@ export const NameValueListWizard = ({ containerRef }) => {
...
-        formValues,
-        setFormValues,
-        id,
-        ref: wizardRef,
-        steps,
-        wizardTitle: 'Wizard Title',
-        width,
-      }}
-    >
+    <WizardContext.Provider value={contextValue}>
```
Comment (halocline) [inline aries-site/src/examples/components/namevaluelist/Examples/NameValueListWizard.js:124]: 👍

File: aries-site/src/examples/templates/FilterControls/FiltersContext.js [hunk 4]
```diff
@@ -26,30 +26,45 @@ const FiltersProvider = ({ children }) => {
...
-    previousFilters,
-    setPreviousFilters,
-    primaryKey,
-    setPrimaryKey,
-    searchValue,
-    setSearchValue,
-    selected,
-    setSelected,
-  };
+  const value = useMemo(
```
Comment (halocline) [inline aries-site/src/examples/templates/FilterControls/FiltersContext.js:29]: 👍

File: aries-site/src/examples/templates/dashboards/DashboardExample.js [hunk 5]
```diff
@@ -1,13 +1,22 @@
...
+    () => ({
+      user,
+      setUser,
+    }),
+    [user],
+  );
+
   return (
-    <UserContext.Provider value={{ user, setUser }}>
+    <UserContext.Provider value={contextValue}>
```
Comment (halocline) [inline aries-site/src/examples/templates/dashboards/DashboardExample.js:19]: 👍  I think this is one of my favorite rules.

File: aries-site/src/examples/templates/dashboards/content/RecentActivity.js [hunk 6]
```diff
@@ -70,7 +70,7 @@ export const RecentActivity = () => {
         (previous, current) => {
           const { title, createdAt, selfUri } = current;
           const next = { ...previous };
-          const count = previous[title]?.count + 1 || 1;
+          const count = previous[title]?.count ? previous[title].count + 1 : 1;
```
Comment (halocline) [inline aries-site/src/examples/templates/dashboards/content/RecentActivity.js:73]: What is this rule? Not sure if I like this one or not.

File: aries-site/src/examples/components/namevaluelist/Examples/NameValueListDefaultExample.js (hunk 2 above)
Comment (taysea) [reply aries-site/src/examples/components/namevaluelist/Examples/NameValueListDefaultExample.js:19]: but in this case I don't think the fragment is even doing anything

File: aries-site/src/examples/templates/dashboards/content/RecentActivity.js (hunk 6 above)
Comment (taysea) [reply aries-site/src/examples/templates/dashboards/content/RecentActivity.js:73]: This rules blocks that the result could be NaN (undefined + 1 maybe?). need to investigate more.

File: aries-site/src/examples/components/namevaluelist/Examples/NameValueListDefaultExample.js (hunk 2 above)
Comment (halocline) [reply aries-site/src/examples/components/namevaluelist/Examples/NameValueListDefaultExample.js:19]: I'm in favor of this rule

File: aries-site/src/examples/templates/dashboards/content/RecentActivity.js (hunk 6 above)
Comment (halocline) [reply aries-site/src/examples/templates/dashboards/content/RecentActivity.js:73]: Was wondering if that what it was. If so, better to keep. I believe we kept it in Grommet.

Comment (taysea) [reply aries-site/src/examples/templates/dashboards/content/RecentActivity.js:73]: Yes we did, I matched Grommet rules.

Comment (britt6612) [reply aries-site/src/examples/templates/dashboards/content/RecentActivity.js:73]: I cant remember if we disabled this rule in grommet or not but it was a strange one

**Review by britt6612 — APPROVED** [review]
everything looks good just the one rule `blocks that the result could be NaN`

File: aries-site/src/examples/templates/dashboards/content/RecentActivity.js (hunk 6 above)
Comment (britt6612) [reply aries-site/src/examples/templates/dashboards/content/RecentActivity.js:73]: oh you answered my question I wasnt sure if we kept it

File: aries-site/src/examples/components/datatable/DataTableMultiSelectExample.js (hunk 1 above)
Comment (taysea) [reply aries-site/src/examples/components/datatable/DataTableMultiSelectExample.js]: Simplified the logic on this one.

**Review by halocline — APPROVED** [review]

**Pull request merged**